src/wd.rs

```diff
@@ -345,3 +345,33 @@ impl TimeoutConfiguration {
         }
     }
 }
+
+/// The response obtained when opening the WebDriver session.
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[non_exhaustive]
+pub struct NewSessionResponse {
+    #[serde(rename = "sessionId")]
+    session_id: String,
+    capabilities: Option<Capabilities>,
+}
+
+impl NewSessionResponse {
+    /// Get the session id.
+    pub fn session_id(&self) -> &str {
+        &self.session_id
+    }
+
+    /// Get the remote end capabilities.
+    pub fn capabilities(&self) -> Option<&Capabilities> {
+        self.capabilities.as_ref()
+    }
+}
+
+impl From<webdriver::response::NewSessionResponse> for NewSessionResponse {
```

Comment:
Unfortunately, even this `impl` isn't okay — it still makes `webdriver` part of the public API. Instead what you'll need is an
```rust
impl NewSessionResponse {
  fn from_wd(nsr: webdriver::response::NewSessionResponse) -> Self
```
so that there are no public `impl`s that name `webdriver`.